r/R/feather.R

```diff
@@ -0,0 +1,160 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+#' @include R6.R
+
+`arrow::ipc::feather::TableWriter` <- R6Class("arrow::ipc::feather::TableWriter", inherit = `arrow::Object`,
+  public = list(
+    SetDescription = function(description) ipc___feather___TableWriter__SetDescription(self, description),
+    SetNumRows = function(num_rows) ipc___feather___TableWriter__SetNumRows(self, num_rows),
+    Append = function(name, values) ipc___feather___TableWriter__Append(self, name, values),
+    Finalize = function() ipc___feather___TableWriter__Finalize(self)
```

Comment:
Is there a rationale for using 3 underscores in these methods like this (instead of 1 or 2)?

Reply:
it's not. There are no convention, and also no way (IIRC) to export a function that lives in a namespace. That's just something semi recognizable and easier to generate automatically if at some point we go that way (à la RcppR6)